docs/end-user/stack-samples-and-call-trees.md

```diff
@@ -0,0 +1,55 @@
+# Stack Samples and Call Trees
+
+The [profiler fundamentals](./profiler-fundamentals) details information about both samples and markers. While samples can collect any type of arbitrary information, this document explores one core idea about samples—that they can collect stacks.
+
+## Stacks gathered over time
+
+![This image shows contiguous stacks, and how they are sampled over time. It fades out as it goes left to right to indicate that samples are gathered for long periods of time. The stacks are labeled by 1 millisecond intervals. The stacks are labeled by functions that are single capital letters, ranged A to H.](../images/samples.svg)
+
+These examples assume the profiler is configured to collect samples every 1 millisecond. As the targeted code runs, the profiler stops the program execution, and samples the stack. Often, the roots of the stacks will look very similar, but then the leaves will vary wildly as the code calls into different parts of the program. If a program is profiled for 3 seconds, at the rate of 1 millisecond per sample, then there would be 3000 samples. It is not practical to view samples individually, so these samples will be aggregated together to provide a more useful look into profile execution. This aggregated data structure is the call tree.
+
+## A simplified call tree
+
+![This image shows stacks going from left to right. They are at 1 millisecond intervals. Each stack is composed of the function, A, B, C, and doWork. There are 5 samples](../images/simple-stacks.svg)
+
+This example creates a fairly contrived and simple example. `A` calls `B`, which calls `C`, then calls `doWork`. `A`, `B`, `C` are very quick to run, and the profiler never directly samples them. However, `doWork` is sampled 5 times. An aggregation of this profile would smush everything into one graph.
+
+![This graphic demonstrates a call tree. It is a chart of the stacks A, B, C, and doWork, each connected by an arrow going from the root A to the leaf doWork. The running time of all the samples is 5ms. The self time is 0ms, except for doWork, which has a self time of 5ms.](../images/simple-call-tree.svg)
+
+This graph is a simplified call tree. Notice that there is no branching, as the samples that are aggregated all share the same stacks. The profiler only samples the function `doWork`, but the rest of the stack is in the report. This means that the running time for every node on the call tree is 5ms. (From here on out a node on a call tree will be referred to as a "call node"). However, since only the `doWork` function was observed, it is the only node to have self time of 5ms, as it was itself observed to be running. `A`, `B`, `C` all have a self time of 0ms.
+
+## Self time in the call tree
```

Comment:
I miss some more explanation of the "running time" before or after this part so that we clearly understand the difference between these 2 values.

(can be added in a subsequent PR)